OpenRA.Mods.Common/Graphics/ContrailRenderable.cs

```diff
@@ -68,22 +74,30 @@ public IRenderable OffsetBy(in WVec vec)
 		public void Render(WorldRenderer wr)
 		{
 			// Note: The length of contrail is now actually the number of the points to draw the contrail
-			// and we require at least two points to draw a tail
+			// and we require at least two points to draw a tail.
 			var renderLength = length - skip;
 			if (renderLength <= 1)
 				return;
 
 			var screenWidth = wr.ScreenVector(new WVec(1, 0, 0))[0];
 			var wcr = Game.Renderer.WorldRgbaColorRenderer;
 
+			var startColor = this.startColor;
+			if (usePlayerStartColor)
+				startColor = Color.FromArgb(this.startColor.A, Player.ActorColor(owner));
```

Comment:
These could use `CachedTransform`s instead.

Reply:
there's no clean way to implement it